tests/samples/test_api.py

```diff
@@ -440,6 +442,40 @@ async def mock_remove_samples(*args, **kwargs):
         assert not m.called
 
 
+@pytest.mark.parametrize("ready", [True, False])
+@pytest.mark.parametrize("exists", [True, False])
+async def test_job_remove(exists, ready, mocker, resp_is, static_time, spawn_job_client, pg_engine):
+    """
+    Test that a sample can be removed when called using the Jobs API.
+
+    """
+    client = await spawn_job_client(authorize=True)
+
+    mocker.patch("virtool.samples.utils.get_sample_rights", return_value=(True, True))
+
+    if exists:
+        file = await virtool.uploads.db.create(pg_engine, "test", "reads", reserved=True)
+
+        await client.db.samples.insert_one({
+            "_id": "test",
+            "all_read": True,
+            "all_write": True,
+            "files": file["id"],
+            "ready": ready
+        })
+
+    mocker.patch("virtool.samples.db.remove_samples", return_value=True)
+
+    resp = await client.delete("/api/samples/test")
+
+    if exists and not ready:
+        assert resp.status == 204
+    elif exists:
+        assert resp.status == 400
+    else:
+        assert resp.status == 404
+
```

Comment:
This test is not effective. Please think again about what behaviours need to be tested to ensure this function works in production.

Also, `remove_samples()` is a coroutine. The `mocker.patch` function only supports synchronous mocks. See https://docs.aiohttp.org/en/stable/testing.html#aiohttp.test_utils.make_mocked_coro.